core/schema/src/Mu/Schema/Conversion/SchemaToTypes.hs

```diff
@@ -45,71 +45,81 @@ typeDefToDecl :: Type -> Namer -> TypeDefB Type String String -> Q [Dec]
 -- Records with one field
 typeDefToDecl schemaTy namer (DRecord name [f])
   = do let complete = completeName namer name
+       fVar <- newName "f"
        d <- newtypeD (pure [])
                      (mkName complete)
-                     []
+                     [PlainTV fVar]
                      Nothing
-                     (pure (RecC (mkName complete) [fieldDefToDecl namer complete f]))
+                     (pure (RecC (mkName complete) [fieldDefToDecl namer complete fVar f]))
                      deriveClauses
-       let hsi = generateHasSchemaInstance schemaTy name complete (fieldMapping complete [f])
+       wTy <- VarT <$> newName "w"
+       let hsi = generateHasSchemaInstance wTy schemaTy name complete (fieldMapping complete [f])
        return [d, hsi]
 -- Records with more than one field
 typeDefToDecl schemaTy namer (DRecord name fields)
   = do let complete = completeName namer name
+       fVar <- newName "f"
        d <- dataD (pure [])
                   (mkName complete)
-                  []
+                  [PlainTV fVar]
                   Nothing
-                  [pure (RecC (mkName complete) (map (fieldDefToDecl namer complete) fields))]
+                  [pure (RecC (mkName complete) (map (fieldDefToDecl namer complete fVar) fields))]
                   deriveClauses
-       let hsi = generateHasSchemaInstance schemaTy name complete (fieldMapping complete fields)
+       wTy <- VarT <$> newName "w"
+       let hsi = generateHasSchemaInstance wTy schemaTy name complete (fieldMapping complete fields)
        return [d, hsi]
 -- Enumerations
 typeDefToDecl schemaTy namer (DEnum name choices)
   = do let complete = completeName namer name
+       fVar <- newName "f"
        d <- dataD (pure [])
                   (mkName complete)
-                  []
+                  [PlainTV fVar]
                   Nothing
                   [ pure (RecC (mkName (choiceName complete choicename)) [])
                     | ChoiceDef choicename <- choices]
                   deriveClauses
-       let hsi = generateHasSchemaInstance schemaTy name complete (choiceMapping complete choices)
+       wTy <- VarT <$> newName "w"
+       let hsi = generateHasSchemaInstance wTy schemaTy name complete (choiceMapping complete choices)
        return [d, hsi]
 -- Simple things
 typeDefToDecl _ _ (DSimple _)
   = fail "DSimple is not supported"
 
 deriveClauses :: [Q DerivClause]
 deriveClauses
-  = [ pure (DerivClause Nothing [ConT ''Eq, ConT ''Ord, ConT ''Show, ConT ''Generic]) ]
+  = [ pure (DerivClause Nothing [{- ConT ''Eq, ConT ''Ord, ConT ''Show, -} ConT ''Generic]) ]
```

Comment:
What is this magic? 💥

Reply:
This is something I need to fix still. Since now turning schemas to types creates a type with a `f (...)` inside, we need to use standalone deriving in order to add an `Applicative f` constraint. I need to figure out how to do this.

Reply:
It turns out that making these instances work correctly would be ton of work, for a module whose use is not recommended. So I've just disabled the generation for now.